lib/services/workflow-api-service.js

```diff
@@ -42,13 +42,17 @@ function workflowApiServiceFactory(
 
     WorkflowApiService.prototype.createAndRunGraph = function(configuration, nodeId) {
         var self = this;
-        return Promise.resolve()
+        return Promise.try(function() {
+            if (!configuration.name || !_.isString(configuration.name)) {
+                throw new Errors.BadRequestError('Graph name is missing or in wrong format');
```

Comment:
maybe it could be 422.  not 400, https://httpstatuses.com/422 

refer to https://developer.github.com/v3/  'Sending invalid fields will result in a 422 Unprocessable Entity response.'